tests/experimental/metric_repository/test_column_filter.py

```diff
@@ -0,0 +1,301 @@
+from __future__ import annotations
+
+import pytest
+
+from great_expectations.core.domain import SemanticDomainTypes
+from great_expectations.experimental.metric_repository.column_filter import ColumnFilter
+from great_expectations.validator.validator import Validator
+
+
+@pytest.fixture
+def mock_validator(mocker):
+    """Create a mock validator for testing."""
+    validator = mocker.Mock(spec=Validator)
+    validator.active_batch_id = "test_batch_id"
+    return validator
+
+
+@pytest.fixture
+def sample_column_names():
+    """Sample column names for testing."""
+    return ["id", "name", "age", "salary", "created_at", "is_active", "description"]
+
+
+@pytest.fixture
+def sample_column_types():
+    """Sample column type information for testing."""
+    return [
+        {"name": "id", "type": "INTEGER"},
+        {"name": "name", "type": "VARCHAR"},
+        {"name": "age", "type": "INTEGER"},
+        {"name": "salary", "type": "DECIMAL"},
+        {"name": "created_at", "type": "TIMESTAMP"},
+        {"name": "is_active", "type": "BOOLEAN"},
+        {"name": "description", "type": "TEXT"},
+    ]
+
+
+class TestColumnFilter:
+    """Test cases for ColumnFilter class."""
+
+    @pytest.mark.unit
+    def test_init_with_defaults(self):
+        """Test initialization with default parameters."""
+        column_filter = ColumnFilter()
+        assert column_filter._include_column_names == []
```

Comment:
This doesn't seem like the right time for me to complain about tests accessing private fields, so I won't.